### PR TITLE
feat: adding json persistance to evaluation

### PR DESCRIPTION
**Pull request by riomus**

### CODE REVIEW

**Review by laugustyniak — APPROVED** [review]
nice :)

**Review by ktagowski — DISMISSED** [review]

File: embeddings/evaluator/evaluator.py [hunk 1]
```diff
@@ -13,3 +15,19 @@ def __init__(self) -> None:
     @abc.abstractmethod
     def evaluate(self, data: Input) -> Output:
         pass
+
+    def persisting(self, persister: ResultsPersister[Output]) -> "Evaluator[Input, Output]":
```
Comment (albert097) [inline embeddings/evaluator/evaluator.py:19]: is it necessary to use quotes? quotes are designed for not defined types https://www.python.org/dev/peps/pep-0484/#forward-references

File: embeddings/utils/json_dict_persister.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+from embeddings.utils.results_persister import ResultsPersister
+
+
+class JsonPersister(ResultsPersister[Dict[str, Any]]):
+    def __init__(self, path: str):
+        self.path = path
+
+    def persist(self, result: Dict[str, Any], **kwargs: Any) -> None:
+        with open(self.path, "w") as json_file:
+            json_file.write(json.dumps(result))
```
Comment (albert097) [inline embeddings/utils/json_dict_persister.py:13]:
```suggestion
            json.dump(result, json_file)
```

or just use `srsly`

**Review by albert097 — CHANGES_REQUESTED** [review]

File: embeddings/evaluator/evaluator.py (hunk 1 above)
Comment (riomus) [reply embeddings/evaluator/evaluator.py:19]: and that is a case, we are referencing to evaluator type in the definition of it

File: embeddings/utils/json_dict_persister.py (hunk 2 above)
Comment (riomus) [reply embeddings/utils/json_dict_persister.py:13]: do we have it already? if we have then for sure, if not I am quite not sure if we should add it as it is not a crucial part, and we will not get any benefits beside increasing the size of the lib

Comment (ktagowski) [reply embeddings/utils/json_dict_persister.py:13]: It was added for promises-election-dataset, but currently I think it is only used in experimental part. Srlsly has nice parser for jsonl format.

Comment (ktagowski) [reply embeddings/utils/json_dict_persister.py:13]: As for now I would accept Albert's suggestion @riomus

**Review by albert097 — APPROVED** [review]

**Review by ktagowski — APPROVED** [review]

**Pull request merged**